src/util/XDRCereal.h

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#ifdef _XDRPP_CEREAL_H_HEADER_INCLUDED_
+#error This header includes xdrpp/cereal.h after necessary definitions \
+       so include this file instead of directly including xdrpp/cereal.h.
+#endif
+
+#include "crypto/Hex.h"
+#include "crypto/SecretKey.h"
+#include "transactions/TransactionUtils.h"
+#include "util/types.h"
+#include <cereal/archives/json.hpp>
+
+using namespace std::placeholders;
+
+template <uint32_t N>
+void
+cereal_override(cereal::JSONOutputArchive& ar, const xdr::opaque_array<N>& s,
+                const char* field)
+{
+    xdr::archive(ar, stellar::binToHex(stellar::ByteSlice(s.data(), s.size())),
+                 field);
+}
+
+// We still need one explicit composite-container override because cereal
+// appears to process arrays-of-arrays internally, without calling back through
+// an NVP adaptor.
+template <uint32_t N>
+void
+cereal_override(cereal::JSONOutputArchive& ar,
+                const xdr::xarray<stellar::Hash, N>& s, const char* field)
+{
+    std::vector<std::string> tmp;
+    for (auto const& h : s)
+    {
+        tmp.emplace_back(
+            stellar::binToHex(stellar::ByteSlice(h.data(), h.size())));
+    }
+    xdr::archive(ar, tmp, field);
+}
+
+template <uint32_t N>
+void
+cereal_override(cereal::JSONOutputArchive& ar, const xdr::opaque_vec<N>& s,
+                const char* field)
+{
+    xdr::archive(ar, stellar::binToHex(stellar::ByteSlice(s.data(), s.size())),
+                 field);
+}
+void
+cereal_override(cereal::JSONOutputArchive& ar, const stellar::PublicKey& s,
+                const char* field)
+{
+    xdr::archive(ar, stellar::KeyUtils::toStrKey<stellar::PublicKey>(s), field);
+}
+
+void
+cereal_override(cereal::JSONOutputArchive& ar,
+                const stellar::MuxedAccount& muxedAccount, const char* field)
+{
+    switch (muxedAccount.type())
+    {
+    case stellar::KEY_TYPE_ED25519:
+        xdr::archive(ar, stellar::KeyUtils::toStrKey(toAccountID(muxedAccount)),
+                     field);
+        return;
+    case stellar::KEY_TYPE_MUXED_ED25519:
+        xdr::archive(ar,
```

Comment:
Nitpick: please don't move code while simultaneously modifying it, because it makes reviewing much more difficult. Can you split this as either "move then modify" or "modify then move" (which is the approach you had previously taken)?

Reply:
Makes sense! Changed it in https://github.com/stellar/stellar-core/pull/2703/commits/a486bd80caae2d8dbf05e116b503f6265695e2fe